src/hugr/rewrite/outline_cfg.rs

```diff
@@ -189,10 +179,36 @@ impl Rewrite for OutlineCfg {
             .ok() // NodePorts does not implement Debug
             .unwrap();
         h.disconnect(exit, exit_port).unwrap();
-        h.connect(exit, exit_port.index(), inner_exit, 0).unwrap();
         // And connect new_block to outside instead
         h.connect(new_block, 0, outside, 0).unwrap();
 
+        // 5. Children of new CFG.
+        let inner_exit = {
+            // These operations do not fit within any CSG/SiblingMut
+            // so we need to access the Hugr directly.
+            let h = h.hugr_mut();
+            let inner_exit = h.children(cfg_node).exactly_one().ok().unwrap();
+            // Entry node must be first
+            h.move_before_sibling(entry, inner_exit).unwrap();
+            // And remaining nodes
+            for n in self.blocks {
+                // Do not move the entry node, as we have already
+                if n != entry {
+                    h.set_parent(n, cfg_node).unwrap();
+                }
+            }
+            inner_exit
+        };
+
+        // 4(b). Reconnect exit edge to the new exit node within the inner CFG
+        let mut in_bb_view: SiblingMut<'_, BasicBlockID> =
```

Comment:
I guess maybe I should comment why we're building these SiblingMut's (i.e. in case the HugrMut we're operating on is a SiblingMut itself), as they would not be necessary if the `h: impl HugrMut` was in fact a `&mut Hugr`